src/main/scala/mesosphere/marathon/api/GroupApiService.scala

```diff
@@ -0,0 +1,72 @@
+package mesosphere.marathon
+package api
+
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.plugin.auth._
+import mesosphere.marathon.raml.{ GroupConversion, Raml }
+import mesosphere.marathon.state._
+
+import scala.async.Async._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GroupApiService(groupManager: GroupManager)(implicit authorizer: Authorizer, executionContext: ExecutionContext) {
+
+  /**
+    * Encapsulates the group update logic that is following:
+    * - if version is set, group is reverted to that version
+    * - if scaleBy is set, group is scaled up
+    * - if none version nor scaleBy is set then the group is updated
+    */
+  @SuppressWarnings(Array("all")) // async/await
+  def updateGroup(
+    rootGroup: RootGroup,
+    groupId: PathId,
+    groupUpdate: raml.GroupUpdate,
+    newVersion: Timestamp)(implicit identity: Identity): Future[RootGroup] = async {
+    val group = rootGroup.group(groupId).getOrElse(Group.empty(groupId))
+
```

Comment:
How about calling `checkAuthorizationOrThrow(UpdateGroup, group)` here? What do you think?